pkg/volume/csi/csi_plugin.go

```diff
@@ -846,6 +852,43 @@ func (p *csiPlugin) getVolumeLifecycleMode(spec *volume.Spec) (storage.VolumeLif
 	return storage.VolumeLifecyclePersistent, nil
 }
 
+// getSupportsFSGroup returns  if the CSI driver supports  supports a volume in the given mode.
+// An error indicates that it isn't supported and explains why.
+func (p *csiPlugin) getSupportsFSGroup(driver string) (storage.SupportsFSGroup, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIVolumeSupportsFSGroup) {
+		// feature is disabled, default to Evaluate
+		return storage.EvaluateSupportsFSGroup, nil
+	}
+
+	// Retrieve CSIDriver. It's not an error if that isn't
+	// possible (we don't have the lister if CSIDriverRegistry is
+	// disabled) or the driver isn't found (CSIDriver is
+	// optional)
+	var csiDriver *storage.CSIDriver
+	if p.csiDriverLister != nil {
+		kletHost, ok := p.host.(volume.KubeletVolumeHost)
+		if ok {
+			if err := kletHost.WaitForCacheSync(); err != nil {
+				return storage.EvaluateSupportsFSGroup, err
+			}
+		}
+
+		c, err := p.csiDriverLister.Get(driver)
+		if err != nil && !apierrors.IsNotFound(err) {
+			// Some internal error.
+			return storage.EvaluateSupportsFSGroup, err
+		}
+		csiDriver = c
+	}
+
+	// If we have an empty SupportsFSGroup we should return the default behavior
+	if csiDriver == nil || csiDriver.Spec.SupportsFSGroup == nil || *csiDriver.Spec.SupportsFSGroup == "" {
```

Comment:
The last 2 cases shouldn't happen, so we should probably return an error instead of assuming an api default.